xbyak/xbyak.h

```diff
@@ -1801,9 +1807,12 @@ class CodeGenerator : public CodeArray {
 	void opJmp(T& label, LabelType type, uint8 shortCode, uint8 longCode, uint8 longPref)
 	{
 		if (isAutoGrow() && size_ + 16 >= maxSize_) growMemory(); /* avoid splitting code of jmp */
+		if (top_ == 0) return;
 		size_t offset = 0;
 		if (labelMgr_.getOffset(&offset, label)) { /* label exists */
-			makeJmp(inner::VerifyInInt32(offset - size_), type, shortCode, longCode, longPref);
+			if (inner::VerifyInInt32(offset - size_) != 0)
+				makeJmp(inner::VerifyInInt32(offset - size_), type, shortCode, longCode, longPref);
```

Comment:
nit: jump with offset 0 is a valid thing, so probably better to check the status instead of not emitting the instruction at all

Reply:
Thanks Changed to checking `IsInInt32()` instead.